sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -554,6 +613,45 @@ def split_next(self):
 
         return left_child_node, right_child_node
 
+    def _compute_interactions(self, node):
+        r"""Compute features allowed by interactions to be inherited by child nodes.
+
+        Example: Assume constraints [{0, 1}, {1, 2}].
```

Comment:
I think this information would be great to have in the User Guide:

https://github.com/scikit-learn/scikit-learn/blob/57e35773f708f6d6129d643a7a237b2fbe0779eb/doc/modules/ensemble.rst?plain=1#L911-L914

What do you think is the best reference for this method?

XREF: I read about the method from the [XGBoost docs](https://xgboost.readthedocs.io/en/latest/tutorials/feature_interaction_constraint.html).

Reply:
I added a user guide entry with https://github.com/scikit-learn/scikit-learn/pull/21020/commits/b9d880bf105f454b1a17789b39e6df3d3dcf802e, but I'm not a Sphinx expert :grimacing:. Help wanted.

Regarding a reference: I'm not aware of a good reference in the literature that explains or depicts it with trees, but the same is true for other parameters. LightGBM and XGBoost are the references for me. While XGBoost has by far the better documentation for interaction constraints, it has/had a wrong implementation for overlapping groups (AFAIK they therefore forbid overlapping groups by now).